low-level/framework/platforms/realstor/realstor_enclosure.py

```diff
@@ -223,28 +227,45 @@ def ws_request(self, url, method, retry_count=MAX_RETRIES,
             if response is None:
                 continue
 
-            if response.status_code != self.ws.HTTP_OK:
-
-                # if call fails with invalid session key request or http 403
-                # forbidden request, login & retry
-                if response.status_code == self.ws.HTTP_FORBIDDEN and relogin is False:
-                    logger.info("%s failed, retrying after login " % (url))
-
-                    self.login()
-                    relogin = True
-                    continue
-
-                elif (response.status_code == self.ws.HTTP_TIMEOUT or \
-                         response.status_code == self.ws.HTTP_CONN_REFUSED or \
-                         response.status_code == self.ws.HTTP_NO_ROUTE_TO_HOST) \
-                         and tried_alt_ip is False:
-                    self.switch_to_alt_mc()
-                    tried_alt_ip = True
-                    self.mc_timeout_counter += 1
-                    continue
-            else:
+            if response.status_code == self.ws.HTTP_OK:
+
                 self.mc_timeout_counter = 0
 
+                try:
+                    jresponse = json.loads(response.content)
+
+                    if jresponse:
+
+                        if jresponse['status'][0]['return-code'] == 2:
```

Comment:
Also please define 
> CLIAPI_RESP_FAILURE = 2 (line 82)
and use it @ line 239
if jresponse['status'][0]['return-code'] == self.CLIAPI_RESP_FAILURE:

Reply:
Done